runtime/src/accounts_db/geyser_plugin_utils.rs

```diff
@@ -95,11 +95,8 @@ impl AccountsDb {
                 match accounts_to_stream.entry(account.meta.pubkey) {
                     Entry::Occupied(mut entry) => {
                         let existing_account = entry.get();
-                        if account.meta.write_version > existing_account.meta.write_version {
-                            entry.insert(account);
-                        } else {
-                            notify_stats.skipped_accounts += 1;
-                        }
+                        assert!(account.meta.write_version > existing_account.meta.write_version);
```

Comment:
How do you guarantee  accounts iterated here are in the increasing order of write_version? If not sure, I think it is simpler to leave this logic unchanged as it works regardless for the same slot.

Reply:
hmm, this is declared in the public interface ReplicaAccountInfoV2 used by plugins, we would need to make that backward compatible when it is removed.

Reply:
@lijunwangs can you explain that, please? What needs to be backwards compatible? The end result should still be the most recent version of each pubkey.

Reply:
The Accounts can be streamed into the storage destination in parallel fashion, the current implementation use the write_version to decide the order when the slot is the same. Like in the postgres plugin. Removing that will break the logic. And ReplicaAccountInfoV2 has that field declared -- we would need to declare a newer version.

Reply:
We can keep the write_version field. It could just always have a value of 0.
I assume we write once a slot becomes a root?
In that case, there will no longer be the same pubkey already in the same slot, so comparisons on write_version become irrelevant.

Reply:
Unfortunately no, we cannot assume the later notification is newer -- it is fundamentally racy when the plugin itself is pushing it into multiple threads. I think the plugin would have to create a write_version itself to maintain the order.

Reply:
> I assume we stream the account into the storage destination once a slot becomes a root?
Is this true?

Reply:
> As long as we stream somewhere near to when a slot becomes a root (basically before we comine into an ancient append vec) there will no longer be the same pubkey already in the same slot, so comparisons on write_version become irrelevant.

2. We can assume going forward that there will not be multiple accounts per pubkey per slot as long as we stream once we become a root. If we stream while txs are storing accounts we could have multiple accounts for one pubkey in a slot.

Reply:
> I think the plugin would have to create a write_version itself to maintain the order.

Can we brainstorm on whether this is feasible? Assuming it is necessary. This depends on the answer to whether we stream at tx store account or once we become a root (or freeze?).

Reply:
> 2\. We can assume going forward that there will not be multiple accounts per pubkey per slot as long as we stream once we become a root. If we stream while txs are storing accounts we could have multiple accounts for one pubkey in a slot.

We are streaming while txns are being applied to the accounts db.

Reply:
> > I think the plugin would have to create a write_version itself to maintain the order.
> 
> Can we brainstorm on whether this is feasible? Assuming it is necessary. This depends on the answer to whether we stream at tx store account or once we become a root (or freeze?).

We stream tx store account time.

Reply:
Note -- we discussed this issue in https://github.com/solana-labs/solana/issues/20910#issuecomment-1082835419

Reply:
ok I removed the geyser part from this pr. We'll deal with it separately.